src/cache.rs

```diff
@@ -6,128 +6,78 @@
-
-static CACHE_DATA_PATH: OnceCell<PathBuf> = OnceCell::new();
-
-fn backend() -> BackendCtor<DiskBackend> {
-    BackendCtor::new(|| {
-        DiskBackend::new(CACHE_DATA_PATH.get().unwrap().join("cache.db"))
-    })
-}
+use rocksdb::{Options, WriteBatch, DB};
 
 /// A cache of notes received from Rusk.
 ///
-/// Before instantiating an instance with [`new`], the cache data path must be
-/// set using [`set_data_path`].
-#[derive(Debug, Clone)]
+/// path is the path of the rocks db database
 pub(crate) struct Cache {
-    data: Map<PublicSpendKey, KeyData>,
+    db: DB,
+    write_batch: WriteBatch,
 }
 
 impl Cache {
-    /// Returns a new cache instance. Before this is called the cache data path
-    /// must be set with [`set_data_path`].
-    ///
-    /// # Panics
-    /// If called before [`set_data_path`].
-    pub(crate) fn new() -> Result<Self, Error> {
-        let data_dir = CACHE_DATA_PATH.get().expect("cache path must be set");
-        let id_path = data_dir.join("cache.id");
-
-        let persisted_id = match id_path.exists() {
-            true => {
-                let bytes = fs::read(id_path)?;
-                let mut source = Source::new(&bytes);
-
-                Some(PersistedId::decode(&mut source)?)
-            }
-            false => None,
-        };
-
-        let data = persisted_id.map_or(Ok(Map::new()), |id| id.restore())?;
-        Ok(Self { data })
-    }
-
-    /// Sets the cache data directory. This function can be called once and is a
-    /// requirement for being able to instantiate a cache using [`new`].
-    ///
-    /// # Panics
-    /// If called more than once or if unable to remove old cache file.
-    pub(crate) fn set_data_path(data_dir: PathBuf) -> Result<(), Error> {
-        // remove old cache if it's still there
-        let db_path = data_dir.join("cache.db");
-        if db_path.exists()
-            && !db_path.is_dir()
-            && fs::remove_file(&db_path).is_err()
-        {
-            panic!("Failed to remove old cache files. Please remove {} manually and try again.", db_path.display())
-        }
-
-        CACHE_DATA_PATH
-            .set(data_dir)
-            .expect("cache path can only be set once");
-
-        Persistence::with_backend(&backend(), |_| Ok(()))?;
-
-        Ok(())
-    }
-
-    /// Persist the cache to the internal backend.
-    pub(crate) fn persist(&self) -> Result<(), Error> {
-        let data_dir = CACHE_DATA_PATH.get().expect("cache path must be set");
-        let id_path = data_dir.join("cache.id");
+    /// Returns a new cache instance.
+    pub(crate) fn new<T: AsRef<Path>>(path: T) -> Result<Self, Error> {
+        let mut opts = Options::default();
+        opts.create_if_missing(true);
+        // After 10 million bytes, sort the cache file and create new one
+        opts.set_write_buffer_size(10_000_000);
 
-        let persistence_id = Persistence::persist(&backend(), &self.data)?;
+        let db = DB::open(&opts, path)?;
```

Comment:
@Daksh14 , I thought the idea is to have a ColumnFamily per a wallet key.  Could you elaborate what data would be retrieved most frequently?

In case we need to retrieve  `note_data` by height, an example schema could be:

`ColumnFamily` per a WalletKey:

    Key pair(s) per each ColumnFamily
    height -> note_encoded_data

Reply:
Using this scheme